config/v1/types_infrastructure.go

```diff
@@ -77,8 +77,32 @@ type InfrastructureStatus struct {
 	// like kubelets, to contact the Kubernetes API server using the
 	// infrastructure provider rather than Kubernetes networking.
 	APIServerInternalURL string `json:"apiServerInternalURI"`
+
+	// ControlPlaneTopology express the expectations for operands that normally run on control nodes.
+	// The default is 'HighlyAvailable', which represents the behavior operators have in a \"normal\" cluster.
+	// The 'SingleReplica' mode will be used in single-node deployments (developer and production) for example,
+	// and the operators should not configure the operand for highly-available operation
+	ControlPlaneTopology TopologyMode `json:"controlPlaneTopology"`
```

Comment:
`// +default=HighlyAvailable`

Reply:
this field is only under the 'status' and not under the 'spec'.
does it need to have a default in the definition here?

Reply:
It does not have to. But a default value shows up automatically if not set otherwise. What should operators do if this field is not set?

Reply:
I see your point, I will add it

Reply:
In the enhancement we said that if the field wasn't set operators should assume `HighlyAvailable` as a default. If setting the default in the CRD definition ensures that a value is provided for clusters upgraded from 4.7 to 4.8 (when the field is added), then maybe we didn't need to worry about that?

Reply:
I'm not sure what will be the value, if at all, on an upgraded cluster.

Reply:
openshift/cluster-config-operator#174 is precedent for updating `Infrastructure.Status` to ensure born-before-the-property clusters are populated the same way that born-now clusters would be populated.  Would also handle "crazy admin manually cleared a `status` property" without requiring all consumers to bake in local default logic, but that seems less likely.  If `// +default=HighlyAvailable` makes some CRD-side tooling handle this defaulting automatically, that makes life easier than teaching the config operator about it.

Reply:
@derekwaynecarr :point_up:

Reply:
I checked it, and I can't use a default value here since this CRD is apiextensions.k8s.io/v1beta1, and default values aren't supported yet.

Reply:
> I checked it, and I can't use a default value here since this CRD is apiextensions.k8s.io/v1beta1, and default values aren't supported yet.

Switch it to v1. We have done that with many CRDs in openshift/api already. No reason not to, just time.

Reply:
@sttts not sure we have extra time to invest into it at this point. do we understand how much effort is it?

Reply:
https://github.com/openshift/api/pull/834

Reply:
Updated the default here and tested the upgrade from an old version. The fields were added to the existing 'cluster' CR with the default value.